python/tests/param/params_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+from six import with_metaclass
+
+from sparkdl.param.converters import SparkDLTypeConverters
+
+from ..tests import PythonUnitTestCase
+
+
+class TestGenMeta(type):
+    """
+    This meta-class add test cases to the main unit-test class.
+    To add test cases, implement the test logic in a function
+    >>> `def _my_test_impl(): ...`
+    then call the following function
+    >>>  `_register_test_case(fn_impl=_my_test_impl, name=..., doc=...)`
+    """
+    def __new__(mcs, name, bases, attrs):
+        _add_invalid_col2tnsr_mapping_tests()
+        attrs.update(_TEST_FUNCTIONS_REGISTRY)
+        return super(TestGenMeta, mcs).__new__(mcs, name, bases, attrs)
```

Comment:
Please do not go meta, see my comment above.

Reply:
Noted, thanks! Please see my response above.